src/lua/LuaShip.cpp

```diff
@@ -1721,6 +1771,7 @@ void LuaObject<Ship>::RegisterClass()
 
 		{ "GetHyperspaceCountdown", l_ship_get_hyperspace_countdown },
 		{ "IsHyperspaceActive", l_ship_is_hyperspace_active },
+		{ "GetGPS", l_get_gps },
```

Comment:
If this function isn't going to be used heavily in a follow-up PR in a way that requires it being implemented in C++, I'd highly prefer implementing it in Lua instead. It's used in only two callsites and is effectively a `ship:GetGroundPosition()` and a `ship:GetVelocityRelTo(ship.frameBody):dot(ship:GetPositionRelTo(ship.frameBody):normalized())` paired together. There's no need to duplicate the same code in the name of a slightly different return parameter interface, especially given this limits the use of GetGPS to a ship object when it can be implemented in terms of only LuaBody API calls.

If you're concerned about the performance of the Get*RelTo functions, you can also expose `body.position` and `body.velocity` as new attributes which would simplify the above to `ship.velocity:dot(ship.position:normalized())`.

Reply:
I'm not concerned with performance, so I used `ship:GetVelocityRelTo(ship.frameBody):dot(ship:GetPositionRelTo(ship.frameBody):normalized())`